### PR TITLE
PR: Migrate to qtpy

### LINKED ISSUE #3101: Migrate to qtpy
Adapt all import to use QtPy instead of Spyder internal Qt Shim

### PR DESCRIPTION
**Pull request by goanpeca**
Fixes #3101

### CODE REVIEW

**Comment by ccordoba12** [comment]
Thanks a lot for doing this @goanpeca! I don't know why but tests are failing intermittently...

**Comment by goanpeca** [comment]
Hmmmm ok, I will keep moving and see how it goes.

**Comment by goanpeca** [comment]
@ccordoba12 tests on Qt5 failed because of [this PR](https://github.com/spyder-ide/qtpy/pull/19)

Is that PR ready to go? can we make a new release of qtpy?

**Comment by goanpeca** [comment]
I tested removing the **spyderlib/qt** to see if everything started. It seems to be working fine.
Should we remove **spyderlib/qt** as part of this PR, or in a separate PR @ccordoba12?

File: spyderlib/widgets/sourcecode/codeeditor.py [hunk 1]
```diff
@@ -8,54 +8,57 @@
...
-                                QTextOption)
-from spyderlib.qt.QtCore import (Qt, Signal, QTimer, QRect, QRegExp, QSize,
-                                 Slot)
-from spyderlib.qt.compat import to_qvariant
-import spyderlib.utils.icon_manager as ima
+# Third party imports
+from qtpy import is_pyqt46
+from qtpy.compat import to_qvariant
+from qtpy.QtCore import QRect, QRegExp, QSize, Qt, QTimer, Signal, Slot
+from qtpy.QtGui import (QBrush, QColor, QCursor, QFont, #QIntValidator,  # FIXME: QTPY needs to be fixed!
```
Comment (goanpeca) [inline spyderlib/widgets/sourcecode/codeeditor.py]: @ccordoba12 please add QIntValidator to the imports (Uncomment) after new QtPy is available.

File: spyderlib/widgets/sourcecode/codeeditor.py [hunk 2]
```diff
@@ -8,54 +8,57 @@
...
+                        QKeySequence, QPaintEvent, QPainter,
+                        QTextBlockUserData, QTextCharFormat, QTextCursor,
+                        QTextDocument, QTextFormat, QTextOption)
+from qtpy.QtPrintSupport import QPrinter
+from qtpy.QtWidgets import (QApplication, QDialog, QDialogButtonBox,
+                            QGridLayout, QHBoxLayout, QInputDialog, QLabel,
+                            QLineEdit, QMenu, QMessageBox, QSplitter,
+                            QTextEdit, QToolTip, QVBoxLayout, QWidget,  # )
+                            QIntValidator)  # FIXME: Remove line after QtPy fix
```
Comment (goanpeca) [inline spyderlib/widgets/sourcecode/codeeditor.py]: @ccordoba12 please remove QIntValidator from the imports (delete line) after new QtPy is available.

**Comment by ccordoba12** [comment]
@goanpeca, I just released a new `qtpy` (1.0.1) with the QIntValidator change. Please do the changes you mentioned yourself, so we don't have to merge a non-passing PR :-)

Thanks!

----

Note: I decided to add this to beta3 because this is ready and also to give our testers the chance to see if there are problems after this change.

**Comment by goanpeca** [comment]
@ccordoba12 :+1:

**Pull request merged**